src/arch/xtensa/CMakeLists.txt

```diff
@@ -462,30 +462,36 @@ else()
 	add_custom_target(run_meu DEPENDS run_rimage)
 endif()
 
+if(NOT DEFINED FIRMWARE_NAME)
```

Comment:
nit-pick: typo on commit title.

Can you also add more explanations on how this would actually be used?

Are you suggesting people manually create new firmware names which don't even have a reference to the platform? e.g. sof-kbl-custom.ri sounds better to me than sof-custom.ri, which risks increasing confusion. I think it's fine to customize, but in addition to a base.

Reply:
Can't the base be incorrect or something? For example imx8 and imx8x could use nearly the same firmware image (and with a modicum of runtime detection the image could be _completely_ shared), so changing the base can be allowed.

Reply:
@plbossart It is for the case when you have exactly the same PCH but for example different MEU config / key. Let's assume that f.e. EHL (that is TGL) will have only different key, then you'd need sof-ehl and sof-tgl binaries. 

Also what do you mean by *base*? `sof-` or `sof-tgl-`, so output binary will be f.e. `sof-tgl-ehl`?

Reply:
@jajanusz ok, but do you mind adding a comment in the code that this is an advanced version and that OSes will need a specific configuration to make use of such custom names?